Extensions/WikiUpdater/WikiUpdaterTask/src/GitWikiFuntions.ts

```diff
@@ -67,21 +67,30 @@ export function GetWorkingFile(filename, logInfo): any {
     return name;
 }
 
-export async function UpdateGitWikiFile(repo, localpath, user, password, name, email, filename, message, contents, logInfo, logError, replaceFile, appendToFile, tagRepo, tag) {
+export async function UpdateGitWikiFile(repo, localpath, user, password, name, email, filename, message, contents, logInfo, logError, replaceFile, appendToFile, tagRepo, tag, injectExtraHeader) {
     const git = simplegit();
 
     let remote = "";
-    let logremote = ""; // make sure we hide the password
-    if (password === null) {
+    let logremote = ""; // used to make sure we hide the password in logs
+    var extraHeaders = [];  // Add handling for #613
+
+    if (injectExtraHeader) {
         remote = `https://${repo}`;
         logremote = remote;
-    } else if (user === null) {
-        remote = `https://${password}@${repo}`;
-        logremote = `https://***@${repo}`;
+        extraHeaders = [`-c http.extraheader="AUTHORIZATION: bearer ${password}"`];
```

Comment:
```suggestion
        extraHeaders = [`-c http.extraheader='AUTHORIZATION: bearer ${password}'`];
```

Reply:
I have swapped to the single quotes for, you can find the build https://dev.azure.com/richardfennell/GitHub/_build/results?buildId=4212&view=results

Another option that might be quicker for you to test is to goto the _work\_tasks folder on your agent, find the task folder and you can hack around with the code. 

Might be a quicker cycle for you to try things out

Reply:
Thank for you for the advice. I do not have my work PC right now. It will have to wait until monday. Have a nice weekend.

Reply:
You too

Reply:
I finally managed to get it to work. It seems like we should not be using neither quotes nor double quotes.
```
extraHeaders = [`-c http.extraheader=AUTHORIZATION: bearer ${password}`];
```
And we do not need to add the extra headers for the pull and push. That's why I deleted my comments.

Reply:
That is good news.

I have made the change to the code, and updated the documentation a bit.

It has just finished building, so you can get te VSIX at https://dev.azure.com/richardfennell/GitHub/_build/results?buildId=4213&view=results 

If that is OK I will release it publicly to the Marketplace, as the other code path (with url credentials) has passed in my release integration tests

Reply:
I just tested the new release. It works perfectly. It is OK for me.

Reply:
Great, in that case I will

- merge the PR
- ship it
- close the issue

Thanks for all the help